File: 1544-make-the-string-great/1544-make-the-string-great.cs
public class Solution {
    public string MakeGood(string s) {
        var stack = new Stack<char>();
        foreach(var ch in s){
            if(stack.Count > 0 && Math.Abs(stack.Peek() - ch) == 32)
                stack.Pop();
            else
                stack.Push(ch);
        }

        var len = stack.Count;
        if(len == 0) return String.Empty;

        var result = new char[len];
        for(var i=len-1; i>=0; i--)
            result[i] = stack.Pop();

        return new String(result);
    }
}

File: 1544-make-the-string-great/1544-make-the-string-great.java
public class Solution {
    public String makeGood(String s) {
        var stack = new Stack<Character>();
        for(var ch : s.toCharArray()){
            if(stack.size() > 0 && Math.abs(stack.peek() - ch) == 32)
                stack.pop();
            else
                stack.push(ch);
        }

        var len = stack.size();
        if(len == 0) return "";

        var result = new char[len];
        for(var i=len-1; i>=0; i--)
            result[i] = stack.pop();

        return new String(result);
    }
}